### ISSUE TITLE
FE_NedelecSZ are not compatible with VectorTools::interpolate

### ISSUE THREAD

**Issue by OresteMarquis** [dealii/dealii#17450]
When FE_NedelecSZ are use to distribute dofs to the DoFHandler it is not possible to use the VectorTools::interpolate function afterwards. 

The resulting error is cryptic :

> An error occurred in line <2791> of file </home/oreste/soft/candi/tmp/unpack/deal.II-v9.5.2/source/fe/fe_values.cc> in function
    dealii::FEValuesBase<dim, spacedim>::FEValuesBase(unsigned int, unsigned int, dealii::UpdateFlags, const dealii::Mapping<dim, spacedim>&, const dealii::FiniteElement<dim, spacedim>&) [with int dim = 3; int spacedim = 3]
The violated condition was: 
    n_q_points > 0
Additional information: 
    There is nothing useful you can do with an FEValues object when using
    a quadrature formula with zero quadrature points!

A simple code reproducing the error is joined (in .txt format so GItHub supports it) below with the CMakeLists:

[CMakeLists.txt](https://github.com/user-attachments/files/16534452/CMakeLists.txt)
[error_c++_file.txt](https://github.com/user-attachments/files/16534422/error_c%2B%2B_file.txt)

**Comment by OresteMarquis** [comment]
@blaisb

**Comment by bangerth** [comment]
Yes, that's clearly not an optimal error message. We should add an assert somewhere higher up in the call stack that checks that what we pass down in information has what the called function actually needs, and if it doesn't provide a better error message. In the current case, this means figuring out why at the point where the `FEValues` object is created, a quadrature formula with zero points is passed in.

@OresteMarquis Want to give it a try identifying where that happens and why?

**Comment by bangerth** [comment]
@OresteMarquis Can you double check that your patch in #17450 fixes the original issue? If so, please close it.

**Comment by bangerth** [comment]
Excellent. Good work, @OresteMarquis !
